oxo.py

```diff
@@ -1,26 +1,48 @@
+#I don't know whats gone wrong...
 class OxoBoard:
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        OxoBoard.board = [[0]*3]*3
+
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return OxoBoard.board[y][x]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if OxoBoard.board[y][x] == 0:
+            OxoBoard.board[y][x] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        for y in OxoBoard.board:
+            for x in y:
+                if x == 0:
+                    return False
+        return True
```

Comment:
Good -- though `x` and `y` are maybe not the best variable names (it's best to only use them when referring to x and y coordinates)